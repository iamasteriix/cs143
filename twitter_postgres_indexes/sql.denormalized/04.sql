/*
SELECT count(*)
FROM tweets 
WHERE to_tsvector('english', text) @@ to_tsquery('english', 'coronavirus') AND lang = 'en';
*/

SELECT count(*)
FROM tweets_jsonb t
WHERE
    to_tsvector('english', coalesce(t.data -> 'extended_tweet' ->> 'full_text', t.data->> 'text')) @@ to_tsquery('english', 'coronavirus')
    AND t.data ->> 'lang' = 'en';

-- bc apparently "tweets" is not a table or a materialized view.
-- once again I am being attacked for presenting new ideas.
