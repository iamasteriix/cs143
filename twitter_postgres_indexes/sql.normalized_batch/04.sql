/*
 * Calculates the hashtags that are commonly used for English tweets containing the word "coronavirus"
 */
SELECT
    count(*)
FROM tweets
WHERE to_tsvector('english',text)@@to_tsquery('english','coronavirus')
  AND lang='en'
;
