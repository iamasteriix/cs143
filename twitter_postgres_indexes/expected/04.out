 count 
-------
  6056
(1 row)

